Assets/Scripts/Doors/DoorAnimator.cs

```diff
@@ -0,0 +1,177 @@
+            rightPanel.localPosition = openRight;
+        }
+    }
+    
+    public void OnTriggerEnter(Collider other)
+    {
+        
+        if (other.gameObject.layer != playerLayer || !isServer) return;
+        
+        if (!insidePlayers.Contains(other.gameObject))
+        {
+            insidePlayers.Add(other.gameObject);
+        }
+    }
+
+    public void OnTriggerExit(Collider other)
+    {
+        if (other.gameObject.layer != playerLayer || !isServer) return;
+        
+        if (insidePlayers.Contains(other.gameObject))
+        {
+            insidePlayers.Remove(other.gameObject);
+        }
+    }
+
+    void OpenDoor()
+    {
+        if (isClosingLocal || isOpeningLocal) return;
+        
+        lerpTime = 0f;
+        StartCoroutine(OpenDoorAnim());
+        openSfx.Play();
+    }
+
+    void CloseDoor()
+    {
+        if (isClosingLocal || isOpeningLocal) return;
+        
+        lerpTime = 0f;
+        StartCoroutine(CloseDoorAnim());
+        closeSfx.Play();
+    }
+
+    IEnumerator OpenDoorAnim()
+    {
+        if(isServer) isOpening = true;
+        isOpeningLocal = true;
+        while (lerpTime < 1f)
+        {
+            lerpTime += Time.deltaTime * (openSpeed * 2f);
+
+            leftPanel.localPosition = Vector3.Lerp(closedLeft, openLeft, lerpTime);
+            rightPanel.localPosition = Vector3.Lerp(closedRight, openRight, lerpTime);
+            yield return new WaitForEndOfFrame();
+        }
+        
+        yield return new WaitForSeconds(1f);
+
+        if(isServer) isOpening = false;
+        isOpeningLocal = false;
+        if(isServer) isOpened = true;
+    }
+
+    IEnumerator CloseDoorAnim()
+    {
+        if(isServer)  isClosing = true;
+        isClosingLocal = true;
+        
+        while (lerpTime < 1f)
+        {
+            lerpTime += Time.deltaTime * openSpeed;
+
+            leftPanel.localPosition = Vector3.Lerp(openLeft, closedLeft, lerpTime);
+            rightPanel.localPosition = Vector3.Lerp(openRight, closedRight, lerpTime);
+            yield return new WaitForEndOfFrame();
+        }
+        
+        yield return new WaitForSeconds(1f);
+        if(isServer) isClosing = false;
+        isClosingLocal = false;
+        if(isServer) isOpened = false;
+    }
+
+    void Update()
+    {
+        if (isServer)
+        {
+            ServerMonitorDoor();
+        }
+        else
+        {
+            ClientMonitorDoor();
+        }
+    }
+
+    void ServerMonitorDoor()
+    {
+        if (insidePlayers.Count > 0 && !isOpening && !isOpened)
+        {
+            OpenDoor();
+        }
+
+        if (insidePlayers.Count == 0 && isOpened)
+        {
+            CloseDoor();
+            isOpened = false;
+        }
+    }
+
+    void ClientMonitorDoor()
+    {
+        if (!isOpeningLocal && isOpening)
+        {
+            OpenDoor();
+        }
+
+        if (!isClosingLocal && isClosing)
+        {
+            CloseDoor();
+        }
+    }
```

Comment:
Instead of isOpeningLocal and isClosingLocal it might be better to use a hook: https://mirror-networking.com/docs/Guides/Sync/SyncVarHook.html

If you attached this hook to isOpened, then you probably don't even need isOpening and isClosing to be networked (you know when they happen because isOpened will change from true to false or false to true)

Reply:
yeah for sure, you guys can refactor it